Doppler.Sap/Services/SapService.cs

```diff
@@ -1,21 +1,130 @@
-using System.Threading.Tasks;
 using Doppler.Sap.Factory;
 using Doppler.Sap.Models;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
 
 namespace Doppler.Sap.Services
 {
     public class SapService : ISapService
     {
         private readonly ISapTaskFactory _sapTaskFactory;
+        private readonly HttpClient _client;
+        private readonly SapConfig _sapConfig;
+        private SapLoginCookies _sapCookies;
+        private DateTime _sessionStartedAt;
 
-        public SapService(ISapTaskFactory sapTaskFactory)
+        public SapService(ISapTaskFactory sapTaskFactory,
+            IOptions<SapConfig> sapConfig,
+            IHttpClientFactory httpClientFactory)
         {
             _sapTaskFactory = sapTaskFactory;
+            _sapConfig = sapConfig.Value;
+            _client = httpClientFactory.CreateClient();
         }
 
         public async Task<SapTaskResult> SendToSap(SapTask dequeuedTask)
         {
             return await _sapTaskFactory.CreateHandler(dequeuedTask);
         }
+
+        public async Task<SapBusinessPartner> TryGetBusinessPartnerByCardCode(string cardCode)
+        {
+            await StartSession();
+
+            var message = new HttpRequestMessage()
+            {
+                RequestUri = new Uri($"{_sapConfig.BaseServerUrl}BusinessPartners('{cardCode}')"),
+                Method = HttpMethod.Get
+            };
+            message.Headers.Add("Cookie", _sapCookies.B1Session);
+            message.Headers.Add("Cookie", _sapCookies.RouteId);
+
+            var sapResponse = await _client.SendAsync(message);
+
+            if (sapResponse.IsSuccessStatusCode)
+            {
+                return JsonConvert.DeserializeObject<SapBusinessPartner>((await sapResponse.Content.ReadAsStringAsync()));
+            }
+            return null;
+        }
+
+        private async Task StartSession()
```

Comment:
This method exists in https://github.com/FromDoppler/doppler-sap/blob/master/Doppler.Sap/Factory/SapTaskHandler.cs#L36 I think the session should be handle by the factory

Reply:
Yes, you're right. As we talk, we took out this logic from SapService.